### PR TITLE
{math,phys}[foss/2021a] deal.II v9.3.3, p4est v2.8

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+configopts = "-DCMAKE_BUILD_TYPE=Release "
+configopts += "-DDEAL_II_WITH_LAPACK=ON -DDEAL_II_WITH_MPI=ON "
+configopts += "-DDEAL_II_WITH_BOOST=ON -DBOOST_DIR=$EBROOTBOOST"
+configopts += "-DDEAL_II_WITH_GSL=ON -DGSL_DIR=$EBROOTBOOST"
+configopts += "-DDEAL_II_WITH_HDF5=ON -DHDF5_DIR=$EBROOTBOOST"
+configopts += "-DDEAL_II_WITH_METIS=ON -DMETIS_DIR=$EBROOTBOOST"
+configopts += "-DDEAL_II_WITH_NETCDF=ON -DNETCDF_DIR=$EBROOTBOOST"
+configopts += "-DDEAL_II_WITH_P4EST=ON -DP4EST_DIR=$EBROOTP4EST"
+configopts += "-DDEAL_II_WITH_PETSC=ON -DPETSC_DIR=$EBROOTBOOST"
```
Comment (branfosj) [inline easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb:37]: This looks wrong. Almost all of those are pointing at `$EBROOTBOOST`. And they are missing a space from the end of the line, so they'll get concatenated into one long string.

Comment (boegel) [reply easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb:37]: Ouch, yeah, I clearly wasn't paying attention much there... Will fix.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3123.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/24788b6715638e8ec3a3f689dea7cb08 for a full test report.

File: easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb]: It looks like you based in on the previous one and that needs correcting as well - see #15692

File: easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    ('HDF5', '1.10.7'),
+    ('METIS', '5.1.0'),
+    ('netCDF', '4.8.0'),
+    ('p4est', '2.8'),
+    ('PETSc', '3.15.1'),
+    ('zlib', '1.2.11'),
+]
+
+configopts = "-DCMAKE_BUILD_TYPE=Release "
+configopts += "-DDEAL_II_WITH_LAPACK=ON -DDEAL_II_WITH_MPI=ON "
```
Comment (branfosj) [inline easybuild/easyconfigs/d/deal.II/deal.II-9.3.3-foss-2021a.eb:30]:
```suggestion
configopts = "-DDEAL_II_WITH_LAPACK=ON -DDEAL_II_WITH_MPI=ON "
```

This is set by the CMake easyblock.

**Review by branfosj — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3547.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/b488c638c791cda4a4eb55d5d3cc3962 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (2 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/0c752b6ac828bc510399708e96325ba4 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (2 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/071df8bb82d7573f1a1270ffcfb51ad8 for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**